common/src/main/java/org/apache/uniffle/common/ShuffleBlockInfo.java

```diff
@@ -95,10 +95,17 @@ public int getLength() {
     return length;
   }
 
-  // calculate the data size for this block in memory including metadata which are
-  // partitionId, blockId, crc, taskAttemptId, length, uncompressLength
+  /**
+   * Calculate the data size for this block in memory including metadata which are partitionId,
+   * blockId, crc, taskAttemptId, uncompressLength and data length. This should be consistent with
+   * {@link ShufflePartitionedBlock#getSize()}.
+   *
+   * @return the encoded size of this object in memory
+   */
   public int getSize() {
-    return length + 3 * 8 + 2 * 4;
+    // FIXME(maobaolong): The size is calculated based on the Protobuf message ShuffleBlock.
+    //  If Netty's custom serialization is used, the calculation logic here needs to be modified.
```

Comment:
Do you mean netty still use the protobuf de/serialization?

Reply:
In fact, Netty need only data size is enough, but for now server release usedmemory contains the  block encoded size

Reply:
> In fact, Netty need only data size is enough

If so, why not removing the unused placeholder bits directly? Just for unified logic for grpc and netty?